promql/prettier/node.go

```diff
@@ -0,0 +1,182 @@
+package prettier
+
+import (
+	"reflect"
+
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+const (
+	grouping = iota
+	scalars
+	multiArguments
+)
+
+type nodeInfo struct {
+	head parser.Expr
+	// Node details.
+	columnLimit int
+	history     []parser.Node
```

Comment:
```suggestion
	ancestors     []parser.Node
```
Just a suggestion. Feel free to ignore.

Reply:
That looks relatable if the function name is updated along with this. So, the function name should go as `nodeAncestors()`?

Reply:
I'm fine with both.

Reply:
`nodeHistory` finding out the `ancestors` makes sense, too.

Reply:
Addressed it.